HTI_2.0.md

```diff
@@ -0,0 +1,451 @@
+
+# Profiles
+
+## FHIR Store profile (HTI:smart_on_fhir)
+The SMART on FHIR profile is elaborated on the following document:
+[HTI on the SMART App Launch Framework](https://docs.google.com/document/d/1qbe2IMIS_zXKMQZY2mQHBUufIHEwMEHo4ESEDmN8P80/edit?usp=sharing)
+
+## JWE message encryption (HTI:jwe)
+In addition to the JWT token, it is also possible to wrap the JWT token in a JWE envelope. The advantage of wrapping the JWT message in a JWE are:
+* The contents of the message can no longer be inspected by the end-user.
+* The message can only be unpacked by the receiver
+
+This adds an extra layer of security over the communication, minimizing the risk of data leak even further.
+
+Disadvantages of using JWE encryption are:
+* Added complexity to the configuration and message creation.
+* Increased difficulty to debug the contents of the JWT message.
+
+#### Restrictions
+The additional requirement for using a JWE message is as follows:
+* The header must contain the key id (kid) of the public key used for encryption.
+
+#### Layout of the message
+For details of the JWE standard, we refer to the JWE documentation. The diagram below displays the JWT and JWE token from a conceptual perspective
+
+![](images/image7.png)
+
+#### Tip: JWT and JWE message detection
+If you, as a module provider, wish to support both JWE and JWT tokens, it is possible to detect id the token is a JWT or JWE token by counting the number of dots in the token. The following rules apply.
+* If the token contains two dots (.), the token is a JWT token.
+* If the token contains four dots (.), the token is a JWT token wrapped in a JWE token.
+
+#### JWE configuration and storage requirements
+The portal needs to configure the following:
+* The public key of the module provider message encryption. The key id (kid) must be part of the token header.
+
+The module provider needs to configure the following
+* The private key for message decryption, referred to by the kid in the message header. The public key of the keypair should be communicated with the module provider for encryption.
+
+![](images/image3.png)
+
+# Implementation checklist
+This section provides an overview of the requirements and responsibilities in Moscow statements of both the portal and module application.
+
+## The portal application
+
+| Description |
+| ------------- |
+| The HTI claims MUST only contain information about the functional task, the definition of the task, and the people involved. |
+| The JWT message MUST only contain information about the sending system, the recipient system and the message itself, with exception to the `sub` field. |
+| The exchange of the message MUST NOT contain any information about: |
+| * the functional task, the definition of the task, and the people involved, and |
+| * information about the sending system, the recipient system and the message itself. |
+| The fields used in the HTI claims object MUST match the table HTI claims field mapping. |
```

Comment:
Er is geen kopje dat "HTI claims field mapping" heet, kan duidelijker zijn waar naar gerefereerd wordt. En "match" is misschien niet de juiste term aangezien er optionele velden zijn. Misschien "conform" ② The message format?